pkg/mounter/refcounter_windows.go

```diff
@@ -0,0 +1,105 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mounter
+
+import (
+	"crypto/md5"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+)
+
+var basePath = "/csi/mounts"
+var mutexes sync.Map
+
+func lock(key string) func() {
+	value, _ := mutexes.LoadOrStore(key, &sync.Mutex{})
+	mtx := value.(*sync.Mutex)
+	mtx.Lock()
+
+	return func() { mtx.Unlock() }
+}
+
+func getRootMappingPath(path string) (string, error) {
+	items := strings.Split(path, "\\")
+	parts := []string{}
+	for _, s := range items {
+		if len(s) > 0 {
+			parts = append(parts, s)
+			if len(parts) == 2 {
+				break
+			}
+		}
+	}
+	if len(parts) != 2 {
+		return "", fmt.Errorf("remote path (%s) is invalid", path)
+	}
+	// parts[0] is a smb host name
+	// parts[1] is a smb share name
+	return strings.ToLower("\\\\" + parts[0] + "\\" + parts[1]), nil
+}
+
+func getMappingPathCount(mappingPath string) int {
+	path := basePath + "/" + strings.Replace(strings.TrimPrefix(mappingPath, "\\\\"), "\\", "/", -1)
+	if os.MkdirAll(path, os.ModeDir) != nil {
+		return -1
+	}
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return -1
+	}
+	count := len(files)
+	return count
+}
+
+func getMd5(path string) string {
+	data := []byte(strings.ToLower(path))
+	return fmt.Sprintf("%x", md5.Sum(data))
+}
+
+func incementMappingPathCount(mappingPath, remotePath string) error {
+	remotePath = strings.TrimSuffix(remotePath, "\\")
+	path := basePath + "/" + strings.Replace(strings.TrimPrefix(mappingPath, "\\\\"), "\\", "/", -1)
+	if err := os.MkdirAll(path, os.ModeDir); err != nil {
+		return err
+	}
+	filePath := path + "/" + getMd5(remotePath)
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		file.Close()
+	}()
+
+	_, err = file.WriteString(remotePath)
+	return err
+}
+
+func decrementMappingPathCount(mappingPath, remotePath string) error {
+	remotePath = strings.TrimSuffix(remotePath, "\\")
+	path := basePath + "/" + strings.Replace(strings.TrimPrefix(mappingPath, "\\\\"), "\\", "/", -1)
+	if err := os.MkdirAll(path, os.ModeDir); err != nil {
+		return err
+	}
+	filePath := path + "/" + getMd5(remotePath)
+	return os.Remove(filePath)
```

Comment:
add a check if current dir is an empty dir, remove whole dir?

Reply:
if `path` dir is empty and `filePath` doesn't exist then `os.Remove(filePath)` returns error. It is expected behaviour.